packages/blocks/src/components/virgo-input/virgo-input.ts

```diff
@@ -0,0 +1,211 @@
+import { VEditor } from '@blocksuite/virgo';
+import * as Y from 'yjs';
+
+import { isDecimal } from './utils.js';
+
+export class VirgoInput {
+  static YTEXT_NAME = 'YTEXT_NAME';
+
+  yDoc: Y.Doc = new Y.Doc();
+  yText: Y.Text;
+
+  vEditor: VEditor;
+
+  readonly type: 'default' | 'number' = 'default';
+  readonly maxLength = Infinity;
+
+  readonly undoManager: Y.UndoManager;
+
+  constructor(options: {
+    rootElement: HTMLElement;
+    yText?: Y.Text | string;
+    maxLength?: number;
+    type?: 'number';
+  }) {
+    const {
+      rootElement,
+      yText = this.yDoc.getText(VirgoInput.YTEXT_NAME),
+      maxLength,
+      type,
+    } = options;
+    const text = yText.toString();
+
+    if (maxLength) {
+      this.maxLength = maxLength;
+      if (text.length > maxLength) {
+        throw 'The text exceeds the limit length.';
```

Comment:
```suggestion
        throw new Error('The text exceeds the limit length.');
```

I think we should aways use this format.